File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        List<ListNode> l = new ArrayList<>();
        ListNode temp = head;
        while(temp!=null){
            l.add(temp);
            temp = temp.next;
        }
        if(l.size()==1){
            return null;
        }
        if(n==l.size()){
            // head = head
            return head.next;
        }
        int index = l.size() - n-1;
        l.get(index).next = l.get(index+1).next;
        return head;
    }
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int index;
    int n;
    ListNode* get(ListNode* head,int size){
        if(head==NULL){
            this->index = size-n;
            return head;
        }
        head->next = get(head->next,size+1);
        if(index==size){
            return head->next;
        }
        return head;
    }
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        this->n = n;
        this->index = 0;
        return get(head,0);
    }
};